src/Package/Impl/ProjectSystem/ProjectProperties.cs

```diff
@@ -92,10 +99,106 @@ public ProjectProperties(ConfiguredProject configuredProject)
+            return remotePath;
+        }
+
+        /// <summary>
+        /// Gets the destination path for a project on the remote host.
+        /// </summary>
+        /// <remarks>
+        /// Default destination is ~/RTVSProjects/.
+        /// </remarks>
+        public async Task SetRemoteProjectPathAsync(string remoteProjectPath) {
+            var runProps = await GetConfigurationRunPropertiesAsync();
+            await runProps.RemoteProjectPath.SetValueAsync(remoteProjectPath);
+        }
+
+        /// <summary>
+        /// Gets the filter string that selects files to be sent to remote host.
+        /// </summary>
+        /// <remarks>
+        /// Default filter string is ".r;.rmd;.rsettings". This selects all R script files, R markdown, 
+        /// and R Settings files.
+        /// </remarks>
+        public async Task<string> GetFileFilterAsync() {
+            var runProps = await GetConfigurationRunPropertiesAsync();
+            var filter = await runProps.TransferFilesFilter.GetEvaluatedValueAsync();
+            if (string.IsNullOrWhiteSpace(filter)) {
+                return ".r;.rmd;.rsettings";
+            }
+            return filter;
+        }
+
+        /// <summary>
+        /// Gets the filter string that selects files to be sent to remote host.
+        /// </summary>
+        /// <remarks>
+        /// Default filter string is ".r;.rmd;.rsettings". This selects all R script files, R markdown, 
+        /// and R Settings files.
+        /// </remarks>
+        public async Task SetFileFilterAsync(string fileTransferFilter) {
+            var runProps = await GetConfigurationRunPropertiesAsync();
+            await runProps.TransferFilesFilter.SetValueAsync(fileTransferFilter);
+        }
+
+        /// <summary>
+        /// Defines if project has to be transfered to remote host during run.
+        /// </summary>
+        public async Task<bool> GetTransferProjectOnRunAsync() {
+            var runProps = await GetConfigurationRunPropertiesAsync();
+            var val = await runProps.TransferProjectOnRun.GetEvaluatedValueAsync();
+            return ParseBooleanProperty(val, false);
+        }
+
+        /// <summary>
+        /// Defines if project has to be transfered to remote host during run.
+        /// </summary>
+        public async Task SetTransferProjectOnRunAsync(bool val) {
+            var runProps = await GetConfigurationRunPropertiesAsync();
+            await runProps.TransferProjectOnRun.SetValueAsync(val);
+        }
+
+        /// <summary>
+        /// Gets all the R script files in the current project.
+        /// </summary>
+        public IEnumerable<string> GetRFilePaths() {
+            IList<string> rFiles = new List<string>();
+            string projectDir = Path.GetDirectoryName(ConfiguredProject.UnconfiguredProject.FullPath);
+            var allFiles = _fileSystem.GetDirectoryInfo(projectDir).GetAllFiles();
+            
+            foreach(var file in allFiles) {
```

Comment:
>foreach [](start = 12, length = 7)

Now that the recursion logic is elsewhere, this loop can become `allFiles.Where(file => file.FullName.EndsWithIgnoreCase(".R"))` (and the caller can call `ToList` or `ToArray` if they need a snapshot, or not if they don't).